ruby-training/trainees/david/test_katas/08/lib/playing_card_deck.rb

```diff
@@ -4,10 +4,23 @@
 class PlayingCardDeck < Deck
   def initialize
     super
-    # your code here
+    @ranks = [*(2..10), 'J', 'Q', 'K', 'A']
+    @suits = ['clubs', 'hearts', 'spades', 'diamonds']
+ 
+    @ranks.each do |rank|
```

Comment:
This is a personal preference, but worth mentioning:

There are about 13 unique ranks and 4 unique suits. Now, imagine you're at the gym. What sounds better, 13 sets of 4 push-ups or 4 sets of 13 push-ups?

I'd personally see 4 sets of 13 as the more logical choice, so in nested loops that create "cartesian products" like this, I like to keep the data structure with the smaller size or more important categories on the outer loop.

So this:

```ruby
@ranks.each do |rank| # Runs 13 times
  @suits.each do |suit| # Runs 4 times
    .
    .
    .
  end
end
```

Becomes this:

```ruby
@suits.each do |rank| # Runs 4 times
  @ranks.each do |suit| # Runs 13 times
    .
    .
    .
  end
end
```